app/controllers/products_controller.rb

```diff
@@ -0,0 +1,49 @@
+class ProductsController < ApplicationController
+  before_action :set_product, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @products = Product.all
+  end
+
+  def show
+  end
+
+  def edit
+  end
+  
+  def new
+    @product = Product.new
+  end
```

Comment:
You can make it look better:
`def edit; end`
`def new; end`

Reply:
Thanks for the suggestion!